tests/mmptest/src/MMPTest.cs

```diff
@@ -813,5 +813,34 @@ public void BuildUnified_ValidSDKDeploymentTargetShouldWork ()
 				TI.TestUnifiedExecutable (test, shouldFail: false);
 			});
 		}
+
+		[Test]
+		[TestCase ("all")]
+		[TestCase ("-all")]
+		[TestCase ("remove-uithread-checks,dead-code-elimination,inline-isdirectbinding,inline-intptr-size,inline-runtime-arch")]
+		public void Optimizations (string opt)
+		{
+			RunMMPTest (tmpDir => {
+				var test = new TI.UnifiedTestConfig (tmpDir) {
+					CSProjConfig = $"<MonoBundlingExtraArgs>--optimize=${opt}</MonoBundlingExtraArgs>",
+				};
+				var rv = TI.TestUnifiedExecutable (test, shouldFail: false);
+				rv.Messages.AssertWarningCount (0);
+				rv.Messages.AssertErrorCount (0);
+			});
+		}
+
+		[Test]
+		public void MM0132 ()
+		{
+			RunMMPTest (tmpDir => {
+				var test = new TI.UnifiedTestConfig (tmpDir) {
+					CSProjConfig = $"<MonoBundlingExtraArgs>--optimize=foo</MonoBundlingExtraArgs>",
+				};
+				var rv = TI.TestUnifiedExecutable (test, shouldFail: false);
+				rv.Messages.AssertWarning (132, "Unknown optimization: 'foo'. Valid optimizations are: remove-uithread-checks, dead-code-elimination, inline-isdirectbinding, inline-intptr-size.");
+				rv.Messages.AssertErrorCount (0);
+			});
+		}
```

Comment:
at some point we'll need to setup something that can share tests, not the the code, between mtouch and mmp

Reply:
Yes, I most certainly agree.